src/hooks/useIntervalFn.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+const useIntervalFn = (fn: any, ms: number) => {
+  const intervalId = useRef<any>()
```

Comment:
intervalId 변수를 보니깐 선언 해주고 밑에서 intervalId 여부에 따라 로직을 체크하는 것이지 다른 용도로 쓰이고 있지가 않은데 그럼 any 타입말고 useRef<HTMLInputElement>(null); 이런식으로 사용해도 되지 않을까요?
혹시 해당 로직에 대해 설명해주실 수 있을까요?

Reply:
useRef를 사용할때는 initvalue로 null을 선언하는게 안전하다고 하네요! 

setInterval로 함수를 정의하면 해당 함수는 정의 된 초기 상태에서 업데이트 되지 않습니다.
가장 대표적인 예제로는 Count를 하나씩 증가시키는 예제가 있습니다.
간략하게 count + 1을 1초 동안 계속 증가시키고자 할때 React에서는 우리가 원하는 대로 1 2 3 증가하지 않습니다.
0 + 1 , 0 +1 , 0 + 1입니다.
이는 setInterval 정의시 count라는 값을 품고있는 콜백함수가 업데이트 되지 않는다는 것을 의미하고 실제로 업데이트하지 않는다고 합니다.
따라서 ref를 사용하여 ref객체의 current 값으로 count를 저장하고 ref 안의 count 값을 기억해서 setInterval 함수 내부 콜백함수가 사용하는 것으로 알고 있습니다.

any 타입으로 하지 않을 경우 intervalId.current Timer형식에 null 형식을 할당할 수 없다는 에러 나옵니다.
setInterval 함수의 반환 값으로 timer 객체가 반환되기 때문에 이 timer 객체에 null, undefined를 할당할 수 없다고 합니다.

setInterval 함수가 NodeJS.Timer 객체를 반환하는 것을 확인하여 아래와 같이 수정하였습니다.
```suggestion
  const intervalId = useRef<NodeJS.Timer>()
```